src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs

```diff
@@ -939,9 +940,16 @@ public CallbackPartition(CancellationTokenSource source)
 
             internal bool Unregister(long id, CallbackNode node)
             {
-                Debug.Assert(id != 0, "Expected non-zero id");
                 Debug.Assert(node != null, "Expected non-null node");
 
+                if (id == 0)
```

Comment:
Not familiar with the calling code, so just curious...

If it's possible for `id == 0 && node == null` (I think this was the case that caused the exception) is it also possible to see `id != 0 && node == null`? What guarantees that?

Reply:
No.  If node is null, we won't ever make it here because this is a method on the node instance.  And the callers that check for node being null don't care about the value of id.  The id only exists to ensure that the holder of the struct has the right to mutate the node instance, and we enter the lock while examining/mutating the node state.